ffi/src/handle.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+use firewood::v2::api::{self, ArcDynDbView, HashKey};
+use metrics::counter;
+
+use crate::DatabaseHandle;
+
+impl DatabaseHandle<'_> {
+    pub(crate) fn get_root(&self, root: HashKey) -> Result<ArcDynDbView, api::Error> {
+        // construct outside of the closure so that when the closure is dropped
+        // without executing, it triggers the hit counter
+        let token = CachedViewHitOnDrop;
+        self.cached_view.get_or_try_insert_with(root, move |key| {
+            token.miss();
+            self.db.view(HashKey::clone(key))
+        })
```

Comment:
nit: this seems unnecessarily complicated, and separates the key names from the function that increments them.

Perhaps the metric management belongs with the ArcCache, since it does seem that any future caller will want a metric around hit rates anyway.

The other choice is a mutable or a second closure to call on hit.

```suggestion
        let mut metric = "firewood.ffi.cached_view.hit";
        self.cached_view.get_or_try_insert_with(root, move |key| {
            metric = "firewood.ffi.cached_view.miss";
            self.db.view(HashKey::clone(key))
        })
        counter!(metric).increment(1);
```

or
```suggestion
        self.cached_view.get_or_try_insert_with(root, move |key| {
            counter!("firewood.ffi.cached_view.miss").increment(1);
            self.db.view(HashKey::clone(key))
        }, |_key| counter!("firewood.ffi.cached_view.hit").increment(1))
```